File: batida_ponto.js
#!/usr/bin/env node

/**
 * Script de automação de batida de ponto - Secullum
 * Usa Playwright para testar login e clique em "Incluir Ponto"
 * Execução: node batida_ponto.js
 */

const { chromium } = require('playwright');
const path = require('path');

// Credenciais
const USUARIO = process.env.BATIDA_USUARIO || '172';
const SENHA = process.env.BATIDA_SENHA || '172';
const URL = 'https://centraldofuncionario.com.br/54128/incluir-ponto';

function log(message) {
  const timestamp = new Date().toLocaleTimeString('pt-BR');
  console.log(`[${timestamp}] ${message}`);
}

async function testarBatidaPonto() {
  log('='.repeat(70));
  log('Iniciando teste de automação - Secullum Batida de Ponto');
  log(`URL: ${URL}`);
  log(`Usuário: ${USUARIO}`);
  log('='.repeat(70));

  let browser;
  let page;

  try {
    // Lançar navegador
    log('Iniciando navegador (Chromium headless)...');
    browser = await chromium.launch({
      headless: true,
      args: ['--disable-blink-features=AutomationControlled']
    });

    const context = await browser.newContext({
      geolocation: undefined,
      permissions: [],
      userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36'
    });

    page = await context.newPage();

    // Acessar URL
    log('Acessando URL...');
    try {
      await page.goto(URL, { waitUntil: 'networkidle', timeout: 15000 });
      log('✅ Página carregada com sucesso');
    } catch (e) {
      log(`⚠️ Timeout ao carregar (${e.message}) - continuando mesmo assim...`);
    }

    // Log de título e informações
    const title = await page.title();
    log(`Título da página: "${title}"`);

    // Procurar e preencher campo de usuário
    log('Procurando campo de usuário...');
    let usuarioField = await page.$('input[name="login"]');
    if (!usuarioField) {
      usuarioField = await page.$('input[type="text"]');
    }

    if (usuarioField) {
      log('✅ Campo de usuário encontrado');
      await usuarioField.fill(USUARIO);
      log(`Usuário '${USUARIO}' preenchido`);
    } else {
      log('❌ Campo de usuário não encontrado');
      const allInputs = await page.$$('input');
      log(`Total de inputs encontrados: ${allInputs.length}`);
      for (let i = 0; i < allInputs.length; i++) {
        const type = await allInputs[i].getAttribute('type');
        const name = await allInputs[i].getAttribute('name');
        const id = await allInputs[i].getAttribute('id');
        log(`  Input ${i}: type="${type}", name="${name}", id="${id}"`);
      }
    }

    // Procurar e preencher campo de senha
    log('Procurando campo de senha...');
    let senhaField = await page.$('input[name="password"]');
    if (!senhaField) {
      senhaField = await page.$('input[type="password"]');
    }

    if (senhaField) {
      log('✅ Campo de senha encontrado');
      await senhaField.fill(SENHA);
      log('Senha preenchida');
    } else {
      log('❌ Campo de senha não encontrado');
    }

    // Procurar e clicar botão de login
    log('Procurando botão de login...');
    let loginButton = await page.$('button:has-text("Entrar")');
    if (!loginButton) {
      loginButton = await page.$('button:has-text("Login")');
    }
    if (!loginButton) {
      loginButton = await page.$('button[type="submit"]');
    }

    if (loginButton) {
      log('✅ Botão de login encontrado');
      await loginButton.click();
      log('Botão de login clicado');

      // Aguardar redirecionamento
      try {
        await page.waitForLoadState('networkidle', { timeout: 10000 });
        log('✅ Página carregada após login');
      } catch (e) {
        log(`⚠️ Timeout ao aguardar carregamento: ${e.message}`);
      }
    } else {
      log('❌ Botão de login não encontrado');
      const allButtons = await page.$$('button');
      log(`Total de botões encontrados: ${allButtons.length}`);
      for (let i = 0; i < allButtons.length; i++) {
        const text = await allButtons[i].textContent();
        log(`  Botão ${i}: "${text.trim()}"`);
      }
    }

    // Aguardar um pouco antes de procurar botão de incluir ponto
    await page.waitForTimeout(2000);

    // Procurar e clicar botão "Incluir Ponto"
    log('Procurando botão "Incluir Ponto"...');
    let incluirButton = await page.$('button:has-text("Incluir Ponto")');
    if (!incluirButton) {
      incluirButton = await page.$('button:has-text("incluir ponto")');
    }
    if (!incluirButton) {
      incluirButton = await page.$('[onclick*="ponto"]');
    }

    if (incluirButton) {
      log('✅ Botão "Incluir Ponto" encontrado');
      await incluirButton.click();
      log('✅✅✅ PONTO BATIDO COM SUCESSO! ✅✅✅');

      await page.waitForTimeout(2000);
      return true;
    } else {
      log('❌ Botão "Incluir Ponto" não encontrado');

      // Verificar se há bloqueio de geolocalização
      const pageText = await page.textContent('body');
      if (pageText.toLowerCase().includes('geolocaliz')) {
        log('⚠️ Possível validação de geolocalização detectada');
      }
      if (pageText.toLowerCase().includes('permiss')) {
        log('⚠️ Possível requisição de permissão detectada');
      }

      // Listar botões disponíveis
      const allButtons = await page.$$('button');
      log(`Total de botões na página: ${allButtons.length}`);
      for (let i = 0; i < Math.min(10, allButtons.length); i++) {
        const text = await allButtons[i].textContent();
        log(`  Botão ${i}: "${text.trim()}"`);
      }

      return false;
    }

  } catch (error) {
    log(`❌ Erro geral: ${error.message}`);
    console.error(error);
    return false;
  } finally {
    if (page) {
      try {
        await page.close();
      } catch (e) {
        // Ignorar erro ao fechar página
      }
    }
    if (browser) {
      try {
        await browser.close();
      } catch (e) {
        // Ignorar erro ao fechar browser
      }
    }
  }
}

// Executar
(async () => {
  try {
    const resultado = await testarBatidaPonto();

    log('='.repeat(70));
    if (resultado) {
      log('✅ TESTE FINALIZADO COM SUCESSO');
      process.exit(0);
    } else {
      log('❌ TESTE NÃO FOI CONCLUÍDO COM SUCESSO');
      process.exit(1);
    }
  } catch (error) {
    log(`❌ Erro fatal: ${error.message}`);
    console.error(error);
    process.exit(1);
  }
})();

File: batida_ponto.py
#!/usr/bin/env python3
"""
Script de automação de batida de ponto - Secullum
Executa login e clica no botão "Incluir Ponto"
"""

import os
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from datetime import datetime

# Credenciais (pode usar variáveis de ambiente para maior segurança)
USUARIO = os.getenv('BATIDA_USUARIO', '172')
SENHA = os.getenv('BATIDA_SENHA', '172')
URL = 'https://centraldofuncionario.com.br/54128/incluir-ponto'

def bater_ponto():
    """Acessa o sistema e bate o ponto"""

    print(f"[{datetime.now().strftime('%H:%M:%S')}] Iniciando batida de ponto...")

    # Configurar opções do Chrome para rodar em background
    options = webdriver.ChromeOptions()
    options.add_argument('--headless')  # Roda sem interface gráfica
    options.add_argument('--no-sandbox')
    options.add_argument('--disable-dev-shm-usage')

    driver = None
    try:
        # Inicializar o navegador
        driver = webdriver.Chrome(options=options)
        driver.get(URL)

        print(f"[{datetime.now().strftime('%H:%M:%S')}] Página carregada")

        # Aguardar e preencher campo de usuário
        campo_usuario = WebDriverWait(driver, 10).until(
            EC.presence_of_element_located((By.NAME, 'login'))
        )
        campo_usuario.send_keys(USUARIO)
        print(f"[{datetime.now().strftime('%H:%M:%S')}] Usuário preenchido")

        # Preencher senha
        campo_senha = driver.find_element(By.NAME, 'password')
        campo_senha.send_keys(SENHA)
        print(f"[{datetime.now().strftime('%H:%M:%S')}] Senha preenchida")

        # Clicar em login
        botao_login = driver.find_element(By.XPATH, "//button[contains(text(), 'Entrar')]")
        botao_login.click()
        print(f"[{datetime.now().strftime('%H:%M:%S')}] Botão de login clicado")

        # Aguardar carregamento da página principal
        WebDriverWait(driver, 15).until(
            EC.presence_of_element_located((By.XPATH, "//button[contains(text(), 'Incluir Ponto')]"))
        )
        print(f"[{datetime.now().strftime('%H:%M:%S')}] Página principal carregada")

        # Clicar no botão "Incluir Ponto"
        botao_incluir = driver.find_element(By.XPATH, "//button[contains(text(), 'Incluir Ponto')]")
        botao_incluir.click()
        print(f"[{datetime.now().strftime('%H:%M:%S')}] ✅ Ponto batido com sucesso!")

        return True

    except Exception as e:
        print(f"[{datetime.now().strftime('%H:%M:%S')}] ❌ Erro: {str(e)}")
        return False

    finally:
        if driver:
            driver.quit()

if __name__ == '__main__':
    bater_ponto()
